### ISSUE TITLE
Version 2.7 will not run on Fedora 40

### ISSUE THREAD

**Issue by dwconsulting** [dangvd/crystal-dock#89]
Looks like you updated to QT 6.8, but Fedora doesn't provide it yet.
(installing rpm from CLI)

"  - nothing provides libQt6Core.so.6(Qt_6.8)(64bit) needed by crystal-dock-2.7-1.x86_64"

**Comment by dangvd** [comment]
Yeah, sorry it's because I built the RPM on Open SUSE Tumbleweed with Qt 6.8 so the RPM will not work on Fedora 40.
The code itself should work so one can still install from the source code.

Also, Fedora 41 should be released this Tuesday with Qt 6.8 so when Fedora users upgrade to Fedora 41, the RPM should work.

I thought about installing Fedora 40 but given the imminent release of Fedora 41, I'm not sure if it's worth it.

**Comment by dwconsulting** [comment]
Code is complaining about a path, didn't have time to diagnose it at the
moment.
I'm good waiting for now, 2.6 still works. Thanks!
(adding cmake error for reference and searching)
-------
❯ cmake -S src -B build -DCMAKE_INSTALL_PREFIX=/usr
-- Configuring done (0.2s)
CMake Error in CMakeLists.txt:
  Imported target "Qt6::GuiPrivate" includes non-existent path

    "/usr/include/qt6/QtGui/6.7.2"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
------

"Since it is so likely that children will meet cruel enemies, let them at
least have heard of brave knights and heroic courage" - C. S. Lewis

**Comment by dangvd** [comment]
Your cmake error was because v2.7 now requires Qt6 Private Dev instead of Qt6 Dev (due to the reworked Auto Hide mode).

I don't know the name of the package in Fedora, but in Open SUSE it's qt6-base-private-devel and in Ubuntu it's qt6-base-private-dev.

**Comment by dwconsulting** [comment]
The package is qt6-qtbase-private-devel

And you are correct, installing that enabled the build.

**Comment by dwconsulting** [comment]
To install them on Fedora 40:

$ sudo dnf install qt6-qtbase-private-devel wayland-devel layer-shell-qt-devel

**Comment by dangvd** [comment]
I've updated README with the info, thanks a lot!

**Comment by dangvd** [comment]
I've added a binary RPM package for Qt 6.7:

https://github.com/dangvd/crystal-dock/releases/tag/v2.7

I built it and tested it on my Fedora 41, and it should also work on Fedora 40.
